### ISSUE TITLE
Publications with only open access main_file_links not being recognized as open access  

### ISSUE THREAD

**Issue by ThriceGood** [LibreCat/LibreCat#418]
I have publications with only main_file_links and no file uploaded locally. Open_access=1 in these publications but they are not being recognized as open access in  the interface. the 'Open Access Publications' link says 0.

**Comment by vpeil** [comment]
Branch https://github.com/LibreCat/LibreCat/compare/pr/issue_418_2nd_edition contains some progress on this issue.

The changes were quite smooth. The only problem remains is the open access filter with the correct number.

Changing the default facets for open_access into
```
open_access:
  terms:
    fields: ['file.open_access', 'main_file_link.open_access']
    size: 1
```
gives on the first a look a correct result. But it seems to count a record with OA file **and** OA link twice.
Any thoughts here?

**Comment by ThriceGood** [comment]
Im trying to add the relevant changes to my layer and I added:

`fields: ['file.open_access', 'main_file_link.open_access']`

and
```
oa:
  op:
     '=': true
     '<>': true
     field: ['main_file_link.open_access', 'file.open_access']
```
Is there anything else I need to change? Because when I restart the application for some reason I see no data at all in the application. 

This error is in the app.log:

```
ERROR [18354] - LibreCat.Search[71] : user->search failed: $VAR1 = {
           'limit' => 20,
           'start' => 0,
           'facets' => {
                         'isi' => {
                                    'terms' => {
                                                 'field' => 'isi',
                                                 'size' => 1
                                               }
                                  },
                         'extern' => {
                                       'terms' => {
                                                    'field' => 'extern',
                                                    'size' => 2
                                                  }
                                     },
                         'pmid' => {
                                     'terms' => {
                                                  'field' => 'pmid',
                                                  'size' => 1
                                                }
                                   },
                         'open_access' => {
                                            'terms' => {
                                                         'field' => [
                                                                      'file.open_access',
                                                                      'main_file_link.open_access'
                                                                    ],
                                                         'size' => 1
                                                       }
                                          },
                         'status' => {
                                       'terms' => {
                                                    'field' => 'status',
                                                    'size' => 8
                                                  }
                                     },
                         'type' => {
                                     'terms' => {
                                                  'size' => 25,
                                                  'field' => 'type'
                                                }
                                   },
                         'popular_science' => {
                                                'terms' => {
                                                             'size' => 1,
                                                             'field' => 'popular_science'
                                                           }
                                              },
                         'year' => {
                                     'terms' => {
                                                  'field' => 'year',
                                                  'order' => 'reverse_term',
                                                  'size' => 100
                                                }
                                   }
                       },
           'sru_sortkeys' => '',
           'cql_query' => 'login=C77F1572-130C-11E8-AFB6-806D539212D2'
         };


```

Any ideas why this might be happening?

Could there just be one field for 'open_access' at the top level of the schema? Because even if you have a main file link and a locally stored file, they would both be open access right? Is it possible have an open access main file link and a non open access local file?

**Comment by vpeil** [comment]
It seems that your changes lead to an elasticsearch query syntax error (have a look at your elasticsearch logs, probably located in /var/log/elasticsearch).

Introducing another top level key **oa** is possible (and is the easiest solution), but maybe not desirable, since we have a lot of keys. 

@phochste what do you think?

**Comment by ThriceGood** [comment]
I managed to get something to work. The error was happening because in the default assets I had 'field' written instead of 'fields'. This is fixed now an I get no elasticsearch error.

At this point I now have the OA facet working in the publication view. But when I click on it I get nothing.

Until I add `main_file_link.open_access` to `publication/cql_mapping/indexes/fulltext` in store.yml

```
           fulltext:
              op:
                '=': true
                'exact': true
                '<>': true
              field: ['main_file_link.open_access', 'file.open_access']
```
Now the open access facet works and returns the publications when clicked on, and also the 'Open Access Publications' link on the home screen displays number of open access publications.

So that works, but like you said, this is still an issue if a publication with both `main_file_link.open_access` and `file.open_access` are counted twice. 


Also I removed what I had originally in store.yml under `publication/cql_mapping/indexes`:

```
            oa:
              op:
                '=': true
                '<>': true
                field: ['main_file_link.open_access', 'file.open_access']
```
I dont know why i put this here, i saw it in the branch you referenced above and thought it was needed.
